homeassistant/components/isy994.py

```diff
@@ -270,6 +300,21 @@ def value(self) -> object:
         # pylint: disable=protected-access
         return self._node.status._val
 
+    def is_unknown(self) -> bool:
+        """Get whether or not the value of this Entity's node is unknown.
+
+        PyISY reports unknown values as -inf
+        """
+        return self.value == -1 * float('inf')
+
+    @property
+    def state(self):
+        """Return the state of the ISY device."""
+        if self.is_unknown():
+            return None
+        else:
+            return super().state
```

Comment:
When would this be used? All isy device classes except binary sensor program seem to override the `state` property anyway. If this call would go to the `Entity` class, it would return `STATE_UNKNOWN`, which would lead to the same outcome as the above `if self.is_unknown()` case.

Reply:
The fan and light isy devices don't implement their own `state` property, so this base-class code would apply to them. I could move the logic in to those last types, but I like that having it in the base class also serves as a signal to future implementers that the UNKNOWN state needs to be handled if you override it.

This actually doesn't end up going to the `Entity` class! I had to test it a bunch because `super()`'s behavior is new to me, but `super` will actually call the implementation in the _other_ base class of the child object.

To use Fan as an example, it is defined as `ISYFanDevice(isy.ISYDevice, FanEntity)`. So `state` will first get handled by `ISYDevice.state`, and when it calls `super()` it kicks it on over to `FanEntity.state` which handles the fan-specific version of state.